### ISSUE TITLE
.rpm, .deb, .msi, etc

### ISSUE THREAD

**Issue by jayvdb** [dephell/dephell#98]
OS distro packaging formats (`.rpm` / `.deb`) and control files (`.spec` & `debian/`) could be another type of hell for dephell to manage ;-)

[bdist_rpm](https://github.com/pypa/setuptools/blob/master/setuptools/command/bdist_rpm.py) is built into setuptools , as a very thin wrapper around distutils, and is still available there.  In Python 3 there is also a .msi builder for Windows, which is probably more sensible than the old `wininst` format.

https://docs.python.org/3/distutils/apiref.html#module-distutils.command

https://github.com/astraw/stdeb/ is bitrotting a little, but there are some semi-active forks.

https://github.com/openSUSE/py2pack/ is a fairly generic .spec generator, with templates designed for openSUSE , and there are a lot of other similar tools and templates for other RPM based systems, so supporting .spec and .srpm shouldnt be too hard.

Is this a direction dephell is interested in pursuing?

**Comment by orsinium** [comment]
Yeah, that's definitely what we have to do :) Also, I have a request from @VadimPushtaev for [rpmvenv](https://github.com/kevinconway/rpmvenv/tree/master) config generation. Thank you for the issue.

**Comment by jhermann** [comment]
Other related tools:

* [spotify/dh-virtualenv: Python virtualenvs in Debian packages](https://github.com/spotify/dh-virtualenv/)
* [jordansissel/fpm: Effing package management! Build packages for multiple platforms (deb, rpm, etc) with great ease and sanity.](https://github.com/jordansissel/fpm)

**Comment by jayvdb** [comment]
Also relevant is https://pypi.org/project/upt/ and its many plugins: https://pypi.org/search/?q=upt
